lib/instrumentation/modules/mongodb.js

```diff
@@ -176,8 +177,11 @@ module.exports = (mongodb, agent, { version, enabled }) => {
 
   function onEnd(event) {
     if (!activeSpans.has(event.requestId)) return;
+    const { failure, reply } = event;
+    const hasFailed = failure || (reply && reply.writeErrors);
     const span = activeSpans.get(event.requestId);
     activeSpans.delete(event.requestId);
+    span.setOutcome(hasFailed ? OUTCOME_FAILURE : OUTCOME_SUCCESS);
```

Comment:
See my comment at https://github.com/elastic/apm-agent-nodejs/issues/3182#issuecomment-1783217513

Reply:
I like the idea of having a specific function fir each type of event. It will prepare the code for the changes you're proposing i the long term.